### ISSUE TITLE
[release] MonoGameExtended 3.6-preview

### ISSUE THREAD

**Issue by lofcz** [craftworkgames/MonoGame.Extended#572]
Any chance that that there will be a prerelease for 2.0? [edit: 3.6 is the next version] Most of us prefer to stick with NuGets instead of building from source. On the other hand, it would be nice for us to have a chance to try new stuff, it's been a long time since 1.1 got released. There's an option for releasing preview / insight build of NuGet package, it would be much appreciated!

**Comment by craftworkgames** [comment]
I'll certainly think about it. There's still a lot of work in progress going on with the 2.0 stuff. There's been a lot of distractions unfortunately.

In the mean time, it's actually very easy to build your own NuGet packages. All you need to do is run the `build.ps1` file and you'll have them.

**Comment by lofcz** [comment]
Cool beans, I'll try that out. If you by any chance have some spare time, could you look into #557? Thanks for your outstanding work.

**Comment by craftworkgames** [comment]
> could you look into #557?

Yep, I looked into it. See the issue for details.

**Comment by craftworkgames** [comment]
> I'd really apprecaite it as well if NuGet would have a more recently updated version. 

That's completely understandable and something I have always intended to do. I don't really have any excuses other than the fact that it's quite difficult to know *when* it makes sense to push a new package. I suppose I should spend some time to think about a policy around this.

One thing I have been thinking about lately along these lines is pushing each individual package separately. For example, I'm currently making some changes to the Entities package and they are mostly isolated to that package. In the past it's been easy enough to push all of the packages at the same time, but these days it's just too hard to keep them all updated in sync.

**Comment by lofcz** [comment]
@craftworkgames I've tried to compile nuget packages from source, yet I keep experiencing errors (cake at first, got that resolved, now something about GitVersion package returning unknown error), it might be something really trivial, any chance I could bother you for a minute of your time somewhere?

**Comment by craftworkgames** [comment]
> now something about GitVersion package returning unknown error

Most likely you didn't clone the git repository using [git](https://git-scm.com/downloads). 

The way GitVersion works, is that it counts the commits and tags on the repository and generates a sensible version number from that. For this to work, it means you need to get the code into an actual git repository on your local machine.

**Comment by lofcz** [comment]
Thanks, got it working. One more question: what happened with Camera2D? No longer it has properties like Position, Zoom and some methods exposed.

**Comment by craftworkgames** [comment]
`Camera2D` has been renamed to `OrthographicCamera` to better reflect what it actually does.

At some point in the future we'll likely introduce a `PerspectiveCamera` as well.

**Comment by KevinYeti** [comment]
@craftworkgames suggest to make a pre-release nuget package at this time. and go on working on that.

**Comment by lofcz** [comment]
Should anyone be interested in nuget build as of 595b9a8bd818bea0ba7daf57b4fb8d8801a033e5: 
http://stagl.cz/fileHosting/artifacts.zip - direct download from my personal blog. 
To install local nupkg files (visual studio): https://stackoverflow.com/questions/10240029/how-do-i-install-a-nuget-package-nupkg-file-locally

**Comment by KevinYeti** [comment]
> Should anyone be interested in nuget build as of [595b9a8](https://github.com/craftworkgames/MonoGame.Extended/commit/595b9a8bd818bea0ba7daf57b4fb8d8801a033e5):
> http://stagl.cz/fileHosting/artifacts.zip - direct download from my personal blog.
> To install local nupkg files (visual studio): https://stackoverflow.com/questions/10240029/how-do-i-install-a-nuget-package-nupkg-file-locally

hi, @lofcz 
Thank you for your pre-release nuget package. And I noticed the version tag is V3.6.0 . 
Is that a right version tag or it should be v2.0 ?

**Comment by craftworkgames** [comment]
> I noticed the version tag is V3.6.0. Is that a right version tag or it should be v2.0 ?

We are using GitVersion to generate the version numbers from the number of commits and branches in the git repository. Essentially, GitVersion counts the commits on each branch and it also counts when branches are merged back into the `develop` branch.

The truth is, I'm not entirely certain [how it works](https://gitversion.readthedocs.io/en/latest/more-info/how-it-works/) but as far as I can tell it's doing the right thing. I agree that version 2.0 "feels" more correct but I fairly strongly believe that it's more correct to let GitVersion do it's thing from a semantic versioning point of view.

Basically, if we fight it now and change it back to 2.0, we'll end up with versions out there that are older but have "newer" version numbers. That's something I'd rather not have to deal with. Let's run just accept that it's version 3.6 and run with it shall we 😉

**Comment by KevinYeti** [comment]
@craftworkgames  Thank you for reply . After I add the package to my local nuget reposity, I realized you are right. The pre-released packages works well now : D

**Comment by KevinYeti** [comment]
The content pipeline seems not working with Tiled file in MGCB. I've add the reference DLL files of the pre-released version to my content manger, but the Importer and Processor are always invalid even can not build the project. 
I've tried to replace the reference files of v1.1 which works OK. Is there any suguestion ?

**Comment by craftworkgames** [comment]
If I recall correctly this is actually an issue with MonoGame 3.7

Try downloading [their latest development build](http://teamcity.monogame.net/repository/download/MonoGame_PackagingWindows/latest.lastSuccessful/MonoGameSetup.exe?guest=1). I believe that will sort it out.

**Comment by KevinYeti** [comment]
> If I recall correctly this is actually an issue with MonoGame 3.7
> 
> Try downloading [their latest development build](http://teamcity.monogame.net/repository/download/MonoGame_PackagingWindows/latest.lastSuccessful/MonoGameSetup.exe?guest=1). I believe that will sort it out.

@craftworkgames 
Thanks for reply. I've tried the development build which version is 3.8.0271. But that did not help.
I add the pre-released version 3.6 of the ME Dlls to the reference, will that a issue of this problem ?

**Comment by craftworkgames** [comment]
Version 3.6.0-beta001 has been [published to NuGet.org](https://www.nuget.org/packages/MonoGame.Extended/3.6.0-beta0001).
